apps/portal/controllers/apis/RESTClientInvoker.jag

```diff
@@ -0,0 +1,310 @@
+     * @param errorCallback    a function to be called if en error is reserved.
+     * @param contentType        requested endpoint should be supported this media type
+     * @param acceptType         this is the media type which is accepted
+     */
+    publicXMLHTTPInvokers.POST = function (url, payload, successCallback, errorCallback, contentType, acceptType) {
+        return privateMethods.XMLHTTPRequest(constants.HTTP_POST, url, payload, successCallback, errorCallback,
+                contentType, acceptType);
+    };
+
+    /**
+     * This method invokes return XMLHTTPRequest for put calls
+     * @param url               target url.
+     * @param payload           payload/data which need to be send.
+     * @param successCallback   a function to be called if the respond if successful.
+     * @param errorCallback     a function to be called if en error is reserved.
+     * @param contentType        requested endpoint should be supported this media type
+     * @param acceptType         this is the media type which is accepted
+     */
+    publicXMLHTTPInvokers.PUT = function (url, payload, successCallback, errorCallback, contentType, acceptType) {
+        return privateMethods.XMLHTTPRequest(constants.HTTP_PUT, url, payload, successCallback, errorCallback,
+                contentType, acceptType);
+    };
+
+    /**
+     * This method invokes return XMLHTTPRequest for delete calls
+     * @param url                target url.
+     * @param successCallback    a function to be called if the respond if successful.
+     * @param errorCallback      a function to be called if en error is reserved.
+     * @param contentType        requested endpoint should be supported this media type
+     * @param acceptType         this is the media type which is accepted
+     */
+    publicXMLHTTPInvokers.DELETE = function (url, successCallback, errorCallback, contentType, acceptType) {
+        var payload = null;
+        return privateMethods.XMLHTTPRequest(constants.HTTP_DELETE, url, payload, successCallback, errorCallback,
+                contentType, acceptType);
+    };
+
+    /**
+     * This method add Oauth authentication header to outgoing HTTPClient Requests if Oauth authentication is enabled.
+     * @param method          HTTP request type.
+     * @param url             target url.
+     * @param payload         payload/data which need to be send.
+     * @param successCallback a function to be called if the respond if successful.
+     * @param errorCallback   a function to be called if en error is reserved.
+     * @param contentType     requested endpoint should be supported this media type
+     * @param acceptType      this is the media type which is accepted
+     */
+    privateMethods.HTTPRequest = function (method, url, payload, successCallback, errorCallback, contentType,
+                                           acceptType) {
+        var HttpClient = Packages.org.apache.commons.httpclient.HttpClient;
```

Comment:
I had a chat with the person who implemented back-end APIs for file downloading in IoTS. So he also tried to use that HTTP client which Jaggery is proving but problem is that client doesn't support the stream processing like file downloading. So that is why he used Apache common HTTP client for that  so we can't change it